src/GovUk.Education.ExploreEducationStatistics.Data.Model/Migrations/Current_Routine_RebuildIndexes.sql

```diff
@@ -110,7 +112,9 @@ BEGIN
 
                     SET @StatsRowIndex += 1;
                 END
-
-            RAISERROR ('Completed updating statistics on tables.', 0, 1, NULL) WITH NOWAIT;
+            IF(@StopTime > GETUTCDATE())
+                RAISERROR ('Completed updating statistics on tables.', 0, 1, NULL) WITH NOWAIT;
+            ELSE
+                RAISERROR ('Reindexing did not complete in %d minutes. Remaining reindexing skipped.', 16, 1, @StopInMinutes) WITH NOWAIT;
```

Comment:
I think there's a case here where all the rebuilding/reorganising of indexes and updating of statistics has completed, but after `StopTime`.

I.e. in the second loop, `StatsRowIndex` is now greater than `StatsCount` so there's no rows left to process, and nothing will be skipped if it were to stop.  The last iteration to be processed was started before `StopTime` but finished after `StopTime`.

If the total run time on completion was over 16 hours, the data factory task will have timed out earlier and we would already have seen a notification like `Data Factory Failure! Stored procedure may still be running!`, which is fine as we may want to manually intervene early.

But if the total run time at this point is between 10 and 16 hours it will fail with a notification `Data factory Failure! Reindexing did not complete in 600 minutes. Remaining reindexing skipped.` giving the impression that there's been a failure and reindexing has been skipped, when reindexing did actually complete in the 16 hours allowed and nothing has been skipped.

If you are bothered about this I think you could make a change to only raise the 'failure' severity level error if processing in either blocks was skipped.